kickoff/static/suggestions.js

```diff
@@ -255,45 +255,60 @@ function getElmoUrl(productName, version) {
+        opts.url = getElmoUrl(productName, version);
+        opts.downloadPlaceholder = 'Elmo';
+    } else {
+        opts.downloadPlaceholder = 'l10n-changesets.json on hg.mozilla.org';
+        var topFolder = productName === 'fennec' ? 'mobile' : 'browser';
+        opts.url = getInTreeFileUrl(branchName, revision, topFolder + '/locales/l10n-changesets.json');
+    }
+
+    opts.previousBuildWarning = '';
+    return opts;
+}
+
+function _getPreviousBuildL10nUrl(productName, version, previousBuildNumber) {
     var baseUrl = window.location.origin;
     var releaseFullName = [productName, version, 'build' + previousBuildNumber].join('-');
     return baseUrl + '/releases/' + releaseFullName + '/l10n';
 }
 
-function getUrlAndMessages(productName, version, buildNumber) {
-    var opts = {
-        url: '',
-        downloadPlaceholder: 'Trying to download from ',
-        previousBuildWarning: '',
-    };
-    var minor = version.match(/(\d+)\.\d+\.\d+/);
-    if (minor) {
+function _getL10nChangesetsOptsOfPreviousBuild(productName, version, buildNumber, minorVersionNumber) {
+    var opts = {};
+    if (minorVersionNumber) {
         // Fall back to $major.0 build1
         var previousBuildNumber = 1;
-        var previousVersion = minor[1] + '.0';
+        var previousVersion = minorVersionNumber[1] + '.0';
+        opts.url = _getPreviousBuildL10nUrl(productName, previousVersion, previousBuildNumber);
         var buildString = previousVersion + ' build' + previousBuildNumber;
-        opts.url = getPreviousBuildL10nUrl(productName, previousVersion, previousBuildNumber);
-        opts.downloadPlaceholder += buildString;
-        opts.previousBuildWarning = 'Changesets copied from ' + buildString +
-            '. If you want to not use them, please edit this field.';
     } else if (buildNumber > 1) {
         var previousBuildNumber = buildNumber - 1;
+        opts.url = _getPreviousBuildL10nUrl(productName, version, previousBuildNumber);
         var buildString = 'build' + previousBuildNumber;
-        opts.url = getPreviousBuildL10nUrl(productName, version, previousBuildNumber);
-        opts.downloadPlaceholder += buildString;
-        opts.previousBuildWarning = 'Changesets copied from ' + buildString +
-            '. If you want to not use them, please edit this field.';
     } else {
-        opts.url = getElmoUrl(productName, version);
-        opts.downloadPlaceholder += 'Elmo';
-        opts.previousBuildWarning = '';
+        throw Error('Unsupported condition for previous build');
     }
 
+    opts.downloadPlaceholder = buildString;
+    opts.previousBuildWarning = 'Changesets copied from ' + buildString +
+        '. If you want to not use them, please edit this field.';
+
     return opts;
 }
 
-function populateL10nChangesets(productName, version, buildNumber) {
+function _getUrlAndMessages(productName, version, buildNumber, branchName, revision) {
+    var minorVersionNumber = version.match(/(\d+)\.\d+\.\d+/);
```

Comment:
The naming is a little confusing - how about majorVersionNumber ?

Reply:
Good call. I think the var was named `minor` which led me to name it this way. That's now fixed.